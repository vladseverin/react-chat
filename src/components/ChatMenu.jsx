import React, { Component } from 'react';
import PropTypes from 'prop-types';
import IconButton from '@material-ui/core/IconButton';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import MoreIcon from '@material-ui/icons/MoreVert';

class ChatMenu extends Component {
  static propTypes = {
    activeUser: PropTypes.shape({
      firstName: PropTypes.string,
      lastName: PropTypes.string,
      username: PropTypes.string,
      isMember: PropTypes.bool.isRequired,
      isCreator: PropTypes.bool.isRequired,
      isChatMember: PropTypes.bool.isRequired,
    }).isRequired,
    disabled: PropTypes.bool.isRequired,
    onLeaveClick: PropTypes.func.isRequired,
    onDeleteClick: PropTypes.func.isRequired,
  };

  state = {
    anchorEl: null,
  };

  handleClick = (event) => {
    this.setState({ anchorEl: event.currentTarget });
  };

  handleClose = () => {
    this.setState({ anchorEl: null });
  };

  handleLeaveClick = () => {
    this.handleClose();
    this.props.onLeaveClick();
  };

  handleDeleteClick = () => {
    this.handleClose();
    this.props.onDeleteClick();
  };

  render() {
    const { activeUser, disabled } = this.props;
    const { anchorEl } = this.state;

    if (!activeUser.isChatMember) {
      return null;
    }

    return (
      <React.Fragment>
        <IconButton
          color="inherit"
          aria-owns={anchorEl ? 'simple-menu' : null}
          aria-haspopup="true"
          disabled={disabled}
          onClick={this.handleClick}
        >
          <MoreIcon />
        </IconButton>
        <Menu
          id="simple-menu"
          anchorEl={anchorEl}
          open={Boolean(anchorEl)}
          onClose={this.handleClose}
        >
          {activeUser.isMember && <MenuItem onClick={this.handleLeaveClick}>Leave</MenuItem>}
          {activeUser.isCreator && <MenuItem onClick={this.handleDeleteClick}>Delete</MenuItem>}
        </Menu>
      </React.Fragment>
    );
  }
}

export default ChatMenu;
