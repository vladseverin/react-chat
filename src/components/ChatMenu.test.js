import React from 'react';
import ReactDOM from 'react-dom';
import ChatMenu from './ChatMenu';

const mockProps = {
  activeUser: {
    firstName: 'firstName',
    lastName: 'lastName',
    username: 'username',
    isMember: true,
    isCreator: true,
    isChatMember: true,
  },
  disabled: false,
  onLeaveClick: jest.fn(),
  onDeleteClick: jest.fn(),
};

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ChatMenu {...mockProps} />, div);
  ReactDOM.unmountComponentAtNode(div);
});
