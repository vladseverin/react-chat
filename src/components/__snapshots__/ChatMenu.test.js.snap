// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatMenu /> renders correctly 1`] = `
<button
  aria-haspopup="true"
  aria-owns={null}
  className="MuiButtonBase-root-33 MuiIconButton-root-27 MuiIconButton-colorInherit-28"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiIconButton-label-32"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-36"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24"
    >
      <g>
        <path
          d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
        />
      </g>
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-42"
  />
</button>
`;
